src/components/ModelingMachineProvider.tsx

```diff
@@ -282,7 +282,7 @@ export const ModelingMachineProvider = ({
         }
         // This DOES NOT set the `selectionRanges` in xstate context
         // same as comment above
-        const selectionRangeTypeMap = dispatchCodeMirrorCursor({
+        const { selectionRangeTypeMap } = dispatchCodeMirrorCursor({
```

Comment:
This was a bug that I'm shocked that typescript didn't pick up on, seems bizarre to me.